### PR TITLE
[FIX][typescript-fetch] Fix self import for models with a discriminator

### PR DESCRIPTION
**Pull request by GeroSchaarmann**
This PR will fix issue #15637 .

In this PR the ```modules/openapi-generator/src/main/resources/typescript-fetch/modelGeneric.mustache``` file was edited, by removing the self import for the discriminator models.

To validate the work use ```./bin/generate-samples.sh bin/configs/typescript-fetch-self-import-issue.yaml``` once with and once without the fix and compare the files it generates.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/typescript-fetch/modelGeneric.mustache [hunk 1]
```diff
@@ -10,14 +10,6 @@ import {
...
 
 {{/hasImports}}
-{{#discriminator}}
-import {
-{{#discriminator.mappedModels}}
-     {{modelName}}FromJSONTyped{{^-last}},{{/-last}}
-{{/discriminator.mappedModels}}
-} from './';
-
-{{/discriminator}}
```
Comment (anton-novikov) [inline modules/openapi-generator/src/main/resources/typescript-fetch/modelGeneric.mustache:20]: I might be horribly wrong here, but this seems to be needed for `legacyDiscriminatorBehavior = true`, however, there is a bug here as well, as `modelName`s might not be unique. Unfortunately, I am not familiar with mustache and have no idea how to fix that

**Comment by Digirik** [comment]
Hello, is this still beeing worked on?

**Comment by Nikos410** [comment]
I just noticed that Gero did not mention the technical committee :)

@TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01) @topce (2018/10) @akehir (2019/07) @petejohansonxo (2019/11) @amakhrov (2020/02) @davidgamero (2022/03) @mkusaka (2022/04)

**Comment by ajansen95** [comment]
Is there still being worked on? I'm having the same issue. If I build my project, I get multiple duplicate errors like this one:
duplicate identifier for legacyDiscriminatorBehavior = false

File: modules/openapi-generator/src/main/resources/typescript-fetch/modelGeneric.mustache (hunk 1 above)
Comment (ajansen95) [reply modules/openapi-generator/src/main/resources/typescript-fetch/modelGeneric.mustache:20]: You might be right on this one. If I check out @GeroSchaarmann 's branch, and set legacyDiscriminatorBehavior=true those imports are missing.
import missing for legacyDiscriminatorBehavior = true

**Comment by macjohnny** [comment]
@GeroSchaarmann do you want to fix the conflicts?

**Comment by Angi2412** [comment]
Any update?
We have the same problem and need to remove the imports manually.

**Comment by joshuamolden** [comment]
Would love to see this go though in the next milestone/version. I am currently having issues with this on a project and would love to not have to modify the generated API code at all.

**Comment by joshuamolden** [comment]
I see this got moved to the next release again. Can this please be resolved in 7.6.0?

**Comment by macjohnny** [comment]
@joshuamolden you can e.g. clone this PR and fix the conflicts to move this forward

**Comment by joshuamolden** [comment]
It seems like the fix would either be:

1. Updating this [modelGeneric.mustache](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/typescript-fetch/modelGeneric.mustache) file to properly account for discriminators being present (since it doesn't seem to be right now)
  * The original author tried to modify this file, but I don't think his changes actually fixed the issue: https://github.com/OpenAPITools/openapi-generator/pull/15638#discussion_r1265614448
2. Controlling the values passed to this mustache file so that it can't make duplicate imports with based on the current mustache template.

I am not sure how to go about option 2, so I think option 1 is most viable.

**Comment by joshuamolden** [comment]
I noticed on my project that I had `legacyDiscriminatorBehavior` set to false. I switched it to true and my code didn't seem to generate with the same issues this ticket is addressing. I would rather not have to opt into the legacy behavior though, so it would still be nice to resolve this ticket.

**Comment by Digirik** [comment]
Due to this PR not being developed any further I forked the branch opened an updatet PR [here](https://github.com/OpenAPITools/openapi-generator/pull/19195).

**Comment by Digirik** [comment]
@macjohnny This can be closed aswell i think. (:

**Pull request closed without merging**